includes/lib/transifex-live-integration-wp-services.php

```diff
@@ -8,11 +8,32 @@
  * Provides WP services to other classes.
  */
 class Transifex_Live_Integration_WP_Services {
+	/**
+	 * Settings array, corresponds to get_option('transifex_live_settings') from DB.
+	 * @var array
+	 */
+	public $settings;
+
+	public function __construct( $settings = null ) {
+		$this->settings = $settings;
+	}
 
   /*
-	 * Wraps WP site_url().
+	 * Wraps WP site_url() / home_url().
+	 * @param bool $is_subdirectory_install Is the site installed in a subdirectory? (see settings defaults)
+	 *           		            					  Use this argument if settings are not available when constructing class.
 	 */
-	function get_site_url() {
-		return site_url();
+	function get_site_url($is_subdirectory_install = null) {
+		$is_subdirectory = false;
+		if (isset($this->settings) && array_key_exists('is_subdirectory_install', $this->settings)) {
+			$is_subdirectory = $this->settings['is_subdirectory_install'];
+		} else if (isset($is_subdirectory_install)) {
```

Comment:
Sorry, will pass on that since what you propose is not equivalent to the original:
- The condition !empty($this->settings['is_subdirectory_install'] && !$is_subdirectory_install) doesn't seem to replicate the original logic correctly. The original function handles $is_subdirectory_install and $this->settings['is_subdirectory_install'] separately to set $is_subdirectory. Your new condition might not produce the same results.
- he code doesn't handle the case when $this->settings['is_subdirectory_install'] is set but false.